src/main/pages/setup-openshift/admin-guide.md

```diff
@@ -157,6 +157,18 @@ See more at [logging docs][logging].
 
 Workspace logs are stored in an PV bound to `che-claim-workspace` PVC. Workspace logs include logs from workspace agent, [boostrapper][boostrapper] and other agents if applicable.
 
+## Che Master States
+
+There is three possible states of the master - `RUNNING`, `PREPARING_TO_SHUTDOWN` and `READY_TO_SHUTDOWN`.
+`PREPARING_TO_SHUTDOWN` state may imply two different behaviors:
+ - When no new workspace startups allowed, and all running workspaces are forcibly stopped;
+ - When no new workspace startups allowed, any workspaces that are currently starting or stopping is allowed to finish that process,
+ and running workspaces doesn't stopped.
+This option is possible only for the infrastructures that support workspaces recovery. For those are didn't, automatic fallback to the shutdown with
+full workspaces stopping will be performed.
+Therefore, `/api/system/stop`  API contract changed slightly - now it tries to do the second behavior by default. Full shutdown with workspaces stop
+can be requested with `shutdown=true` parameter.
+
```

Comment:
What is READY_TO_SHUTDOWN and how differs from PREPARING_TO_SHUTDOWN state?
As I understand it when you perform "stop" the system:
- tries to stop workspaces gracefully (if infra supports recovery)
- otherwise it just stopping workspaces
that's PREPARING_TO_SHUTDOWN state?
when all stopped it becomes READY_TO_SHUTDOWN which allows master to stop the process?

Reply:
Yes, that's correct.

Reply:
Added  sentence when PREPARING_TO_SHUTDOWN becomes READY_TO_SHUTDOWN